### PR TITLE
Vasilev Ivan. Lab2. Var 3

### PR DESCRIPTION
**Pull request by AdAndRoll**

### CODE REVIEW

File: llvm/labs/lab2/vasilev_ivan/Vasilev.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+      if (!Preheader || ExitBlocks.empty())
+        continue;
+
+      int loop_start = 0;
+      int loop_end = 0;
+
+      for (Instruction &Instr : *Preheader) {
+        if (CallInst *CI = dyn_cast<CallInst>(&Instr)) {
+          if (CI->getCalledFunction() &&
```
Comment (m-ly4) [inline llvm/labs/lab2/vasilev_ivan/Vasilev.cpp]: Please use searching in loop_start/loop_end funcs users lists for this check.

**Review by m-ly4 — CHANGES_REQUESTED** [review]

File: llvm/labs/lab2/vasilev_ivan/Vasilev.cpp (hunk 1 above)
Comment (AdAndRoll) [reply llvm/labs/lab2/vasilev_ivan/Vasilev.cpp]: I'm sorry, I'm not quite sure what you mean.

Comment (m-ly4) [reply llvm/labs/lab2/vasilev_ivan/Vasilev.cpp]: Please, look into `users()` method of Value class. https://llvm.org/doxygen/classllvm_1_1Value.html#a411cf3e3932f209ce3374cb31adc1da6

File: llvm/test/lab2/vasilev_ivan/test.ll [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+; CHECK-NOT:    call void @_Z8loop_endv()
+  ret void
+}
+
+define dso_local void @_Z15with_lopps_funcv() {
+entry:
+  %i = alloca i32, align 4
+  store i32 0, ptr %i, align 4
+; CHECK:    call void @_Z10loop_startv()
+; CHECK-NOT:    call void @_Z10loop_startv()
```
Comment (m-ly4) [inline llvm/test/lab2/vasilev_ivan/test.ll]:
```suggestion
  store i32 0, ptr %i, align 4
  call void @_Z10loop_startv()
; CHECK:    call void @_Z10loop_startv()
; CHECK-NOT:    call void @_Z10loop_startv()
```

File: llvm/test/lab2/vasilev_ivan/test.ll [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+while.body:
+  %1 = load i32, ptr %i, align 4
+  %inc = add nsw i32 %1, 1
+  store i32 %inc, ptr %i, align 4
+  br label %while.cond
+
+while.end:
+; CHECK:    call void @_Z8loop_endv()
+; CHECK-NOT:    call void @_Z8loop_endv()
+  ret void
```
Comment (m-ly4) [inline llvm/test/lab2/vasilev_ivan/test.ll]:
```suggestion
while.end:
  call void @_Z8loop_endv()
; CHECK:    call void @_Z8loop_endv()
; CHECK-NOT:    call void @_Z8loop_endv()
  ret void
```

**Review by m-ly4 — CHANGES_REQUESTED** [review]

**Review by m-ly4 — APPROVED** [review]

**Pull request merged**